src/NServiceBus.Hosting.Windows/StartableAndStoppable/StartableAndStoppableRunner.cs

```diff
@@ -0,0 +1,103 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Hosting;
+    using Logging;
+
+    class StartableAndStoppableRunner
+    {
+        public StartableAndStoppableRunner(IEnumerable<IWantToRunWhenEndpointStartsAndStops> wantToRunWhenBusStartsAndStops)
+        {
+            this.wantToRunWhenBusStartsAndStops = wantToRunWhenBusStartsAndStops;
+        }
+
+        public Task Start(IMessageSession session)
+        {
+            var startableTasks = new List<Task>();
+            foreach (var startable in wantToRunWhenBusStartsAndStops)
+            {
+                var task = startable.Start(session).ThrowIfNull();
+
+                var startable1 = startable;
+
+                /* 
+                    We can't use the await keyword because of the conditional logging. 
+                    Since we want to start them concurrently and log per instance there is not much else we can do.
+                */
+                task.ContinueWith(t =>
+                {
+                    thingsRanAtStartup.Add(startable1);
+                    Log.DebugFormat("Started {0}.", startable1.GetType().AssemblyQualifiedName);
+                }, TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously);
+                task.ContinueWith(t =>
+                {
+                    Log.Error($"Startup task {startable1.GetType().AssemblyQualifiedName} failed to complete.", t.Exception);
+                }, TaskContinuationOptions.OnlyOnFaulted | TaskContinuationOptions.ExecuteSynchronously);
+                
+                startableTasks.Add(task);
+            }
+
+            return Task.WhenAll(startableTasks.ToArray());
```

Comment:
Since this blocks startup if users do a "while(true)" in their start runner should we add a timeout task to at least do some logging that "this startup runner takes a long time to start"? (cc @indualagarsamy )

Also we need to spell this out very clear in API doco that user should not do any blocking stuff in there. I v5 this was ok since we allowed ".Start" to never complete and users porting their code to v6 will hit this and their endpoints will hang on startup.